client/blocks/reader-suggested-follows/hooks/use-recommend-or-related-sites-query/index.ts

```diff
@@ -0,0 +1,72 @@
+import {
+	FeedRecommendation,
+	useFeedRecommendationsQuery,
+} from 'calypso/data/reader/use-feed-recommendations-query';
+import { RelatedSite, useRelatedSites } from 'calypso/data/reader/use-related-sites';
+
+interface Author {
+	wpcom_login: string;
+	ID: string;
+	name: string;
+}
+
+interface QueryParams {
+	author?: Author;
+	siteId: number;
+	postId: number;
+}
+
+interface QueryOptions {
+	enabled: boolean;
+}
+
+const getResourceType = (
+	recommendedFeeds: FeedRecommendation[] | null | undefined,
+	relatedSites: RelatedSite[] | null | undefined
+) => {
+	if ( Array.isArray( recommendedFeeds ) && recommendedFeeds?.length > 0 ) {
+		return 'recommended';
+	}
+
+	if ( Array.isArray( relatedSites ) && relatedSites?.length > 0 ) {
+		return 'related';
+	}
+
+	return null;
+};
+
+export const useRecommendOrRelatedSitesQuery = ( query: QueryParams, options?: QueryOptions ) => {
+	const { author, siteId, postId } = query;
+	const userLogin = author?.wpcom_login || author?.ID;
+	const enabled = options?.enabled ?? true;
+
+	const {
+		data: recommendedFeeds,
+		isLoading: isLoadingRecommendedFeeds,
+		isSuccess: isSuccessRecommendedFeeds,
+	} = useFeedRecommendationsQuery( userLogin, {
+		enabled: !! userLogin && enabled,
+	} );
+
+	const hasRecommendedFeeds = recommendedFeeds && recommendedFeeds.length > 0;
+	const shouldLoadRelatedSites = enabled && isSuccessRecommendedFeeds && ! hasRecommendedFeeds;
+
+	const {
+		data: relatedSites,
+		isLoading: isLoadingRelatedSites,
+		isSuccess: isSuccessRelatedSites,
+	} = useRelatedSites( siteId, postId, {
+		enabled: shouldLoadRelatedSites,
+	} );
+
+	const data = recommendedFeeds?.length > 0 ? recommendedFeeds : relatedSites;
+	const isLoading = isLoadingRecommendedFeeds || isLoadingRelatedSites;
+	const isSuccess = isSuccessRecommendedFeeds || isSuccessRelatedSites;
```

Comment:
```suggestion
	const isSuccess = ( isSuccessRecommendedFeeds && hasRecommendedFeeds ) || isSuccessRelatedSites;
```

Would we want to check if we have a recommended feed here as well? Otherwise we will return `true` before we even run useRelatedSites?

Reply:
I don't think so, because we can request with success + empty recommended list, which is still considered success. 
The component can decide how to handle this scenario.

Do you have any case in mind?

Reply:
What would be the result if we get a successful but empty result from `useFeedRecommendationsQuery` then we get an unsuccessful from `useRelatedSites`. What should the hook return in that case? As a consumer of the hook I think I would expect `isSuccess` to be false since the first query was empty and the fallback was not successful.

Haha maybe my solution doesn't work fully either. Just throwing out ideas 😄

Reply:
I can see...

It is a puzzle 😨  and also related to your other comment https://github.com/Automattic/wp-calypso/pull/104927/files#r2242766457.

Let me think a bit about how to solve it.

Reply:
Thinking a bit more abou

When we think about fallback, we need to return the best answer we have at the moment, or a third fallback strategy to some hardcoded data that I guess is not viable here.  (Please let me know if you have any idea how to do this.) 

On the scenario you are describing, the best answer we have is "our requests were finished and we don't have a recommended list of items for you," which enables the frontend to show any message or just skip the rendering. 

In this context, the 'isSuccess' means more the work to get the data done instead of  "we have data for you". 

Thinking from the user perspective, doesn't matter much if we don't have data because we don't have it or because one request returned success and the fallback failed, the final state should be not mess the client. 

Maybe we can do here is add a tracking to check how frequently this scenario is happening and reevaluate the solution. 


What do you think?

Reply:
No sweat at all, I don't think it's a high probability issue. No need to add tracking unless you are wanting to chase it down. I was just mentioning it in the chance that this hook gets used elsewhere. What you got should be fine unless you see otherwise 👍